verify-rpms/rpm_verifier.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+
+from pathlib import Path
+from typing import Callable
+
+import click
+
+
+def get_rpmdb(container_image: str) -> Path:
+    raise NotImplementedError()
+
+
+def get_rpms(rpmdb: Path) -> list[str]:
+    raise NotImplementedError()
+
+
+def get_unsigned_rpms(rpms: list[str], verifier: Callable[[str], bool]) -> list[str]:
+    return [rpm for rpm in rpms if verifier(rpm)]
+
+
+def is_rpm_unsigned(rpm: str) -> bool:
+    raise NotImplementedError()
+
+
+def generate_output(unsigned_rpms: dict[str, list[str]]) -> bool:
+    raise NotImplementedError()
+
+
+@click.command()
+@click.option(
+    "--container-image",
+    help="Reference to container image",
+    type=str,
+    multiple=True,
+)
+def main(container_image: list[str]):
```

Comment:
I don't think its beneficial to type every varaible. The variable typing syntax was really only meant for typing class members and for when the type inference can't figure it out.

Note that even fully statically typed languages opt out of explicitly typing individual variables these days.

Reply:
having it made it easier for me to write the dict comprehensions.
Do you think it will make the code more readable at this stage if it stays or if it goes?

Reply:
I think it would be easier to read if it goes.

One thing that occurred to me is that I'd rather have generators of pairs there instead of dicts, but that is an implementation detail we can tend to later. We do need to make sure we structure the tests to enable such a re-factor.

Reply:
I'll remove it.